R/app.R

```diff
@@ -334,16 +334,21 @@ initAutoReloadMonitor <- function(dir) {
 #'   `NULL`, `global.R` will not be evaluated at all.
 #' @export
 loadSupport <- function(appDir=NULL, renv=new.env(parent=globalenv()), globalrenv=globalenv()){
+  require(shiny)
+
+  if (is.null(appDir)) {
+    appDir <- findEnclosingApp(".")
+  }
+
   if (!is.null(globalrenv)){
     # Evaluate global.R, if it exists.
     if (file.exists(file.path.ci(appDir, "global.R"))){
-      sourceUTF8(file.path.ci(appDir, "global.R"), envir=globalrenv)
+      withr::with_dir(appDir, {
+        sourceUTF8("global.R", envir=globalrenv)
```

Comment:
I think this needs `file.path.ci` in order to work on systems with a case-sensitive filesystem where global.R has a different capitalization, like Global.r. Should be able to test this out on Linux.

Reply:
Thanks, got it. Also applied to `findEnclosingApp()`